Bug: Types used in JSDoc blocks of @eslint/js are missing

### Environment

Node version: v21.2.0
npm version: v10.2.3
Local ESLint version: v8.56.0 (Currently used)
Global ESLint version: Not found
Operating System: NixOS 23.11 (Tapir)

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

`packages.json`:
```json
{
  "main": "index.js",
  "type": "module",
  "scripts": {
    "lint": "npx eslint && npx tsc -p jsconfig.json"
  },
  "devDependencies": {
    "@eslint/js": "^8.56.0",
    "eslint": "^8.56.0",
    "typescript": "^5.3.3"
  }
}
```
`index.js`:
```js
import "@eslint/js";
```
`jsconfig.json`:
```json
{
  "compilerOptions": {
    "checkJs": true
  },
  "exclude": ["node_modules"]
}
```

</details>

Running `npx tsc -p jsconfig.json` complains about missing types used by `@eslint/js`

### What did you expect to happen?

No missing type errors caused by `@eslint/js`

### What actually happened?

```bash
❯ npx tsc -p jsconfig.json
node_modules/@eslint/js/src/configs/eslint-recommended.js:11:19 - error TS2307: Cannot find module '../lib/shared/types' or its corresponding type declarations.

11 /** @type {import("../lib/shared/types").ConfigData} */
                     ~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@eslint/js/src/configs/eslint-recommended.js:11


```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/eslintjs-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

_No response_

> @colireg Would you like to send a PR?

I tried to look into it for a while, but since this is a separate package, fixing this issue is far trickier than I initially thought. I'm not very experienced with npm package publishing, but I'm pretty sure simply updating the path in the comment wouldn't fix it. I tried to add `eslint` as a dependency for `@eslint/js` and import types directly from it, but it led to another type error:
```bash
../../../git/eslint/lib/shared/types.js:77:15 - error TS2304: Cannot find name 'ScopeManager'.

77  * @property {ScopeManager} [scopeManager] The scope manager of the AST.
                 ~~~~~~~~~~~~


Found 1 error in ../../../git/eslint/lib/shared/types.js:77
```
I tried to look for `ScopeManager` definition in the code base to no avail. Removing the line 77 seems to fix the issue, but I don't think that's a solution. Guess at this point I can't do anything useful except removing the JSDoc comment alltogether ;)

--- a/packages/js/src/configs/eslint-recommended.js
+++ b/packages/js/src/configs/eslint-recommended.js
@@ -8,7 +8,6 @@
 
 /* eslint sort-keys: ["error", "asc"] -- Long, so make more readable */
 
-/** @type {import("../lib/shared/types").ConfigData} */
 module.exports = Object.freeze({
     rules: Object.freeze({
         "constructor-super": "error",

